emcc.py

```diff
@@ -46,9 +46,10 @@
 CXX_ENDINGS = ('.cpp', '.cxx', '.cc', '.c++', '.CPP', '.CXX', '.CC', '.C++', '.ii')
 OBJC_ENDINGS = ('.m', '.mi')
 OBJCXX_ENDINGS = ('.mm', '.mii')
+EC_ENDINGS = ('.ec',)
```

Comment:
what is an `.ec` file?

Reply:
eC language ( http://ec-lang.org )
Alternatively, perhaps the default should NOT be C++?

Reply:
We have the C and C++ prefixes etc. here because we support them as inputs - we send the files to clang, and it knows what to do with them. If clang doesn't support eC, then I don't think there's a benefit to adding the prefix here?

Reply:
The eC compiler tools support Emscripten / Binaryen as a target.
The compiler does a first preprocessing pass by invoking the C preprocessor, and Clang can preprocess eC source files just fine. Because emcc defaults to C++ for unknown extensions (wrongly, in my opinion) it always breaks the builds because it specifies a C++ standard by default.

I have raised this issue in the past and it always keeps coming back. Please adopt either of the suggested fixes to not automatically treat either unknown extension, or specifically eC source files, as C++.

This issue was made worst with the explicit '-x c' not working.